src/Interpreters/ActionsVisitor.cpp

```diff
@@ -64,6 +64,16 @@ static NamesAndTypesList::iterator findColumn(const String & name, NamesAndTypes
                         [&](const NamesAndTypesList::value_type & val) { return val.name == name; });
 }
 
+static size_t getTypeDepth(const DataTypePtr & type)
```

Comment:
There is no check for stack overflow neither the comment about why stack overflow does not happen.

I will add a check.